src/renderers/dom/client/ReactMount.js

```diff
@@ -880,9 +880,25 @@ var ReactMount = {
           checksum
         );
 
-        var diffIndex = firstDifferenceIndex(markup, rootMarkup);
+        // because rootMarkup is retrieved from the DOM, various normalizations
+        // will have occurred which will not be present in `markup`. Here,
+        // insert markup into a <div> or <iframe> depending on the container
+        // type to perform the same normalizations before comparing.
+        var normalizer, normalizedMarkup;
+        if (container.nodeType === ELEMENT_NODE_TYPE) {
+          normalizer = document.createElement('div');
+          normalizer.innerHTML = markup;
```

Comment:
I'd prefer to just perform this check in dev mode, that was certainly something I was anticipating :)